src/python/WMCore/MicroService/Unified/MSRuleCleaner.py

```diff
@@ -3,22 +3,42 @@
 
 Description:
 This MicroService is meant to remove Rucio rules that are no longer needed in the Workload Management system, such as:
- * block-level rules created by WMAgent, against the origin RSE where data is getting produced
- * container and block-level rules created by MSTransferor, for input data that is no longer in the system
+ * block-level     rules created by WMAgent, against the origin RSE where data is getting produced
+ * container-level rules created by WMAgent, with a very generic RSE expression
+ * block-level     rules created by MSTransferor, for input data that is no longer in the system
+ * container-level rules created by MSTransferor, for input data that is no longer in the system
 In addition to that, this MicroService is now also responsible for the workflow archival, which is the final status that
 workflows remain.
 """
+# TODO: To Add the top level description of the logic into the module's Docstring
 
 # futures
 from __future__ import division, print_function
+import json
 
 # system modules
-from retry import retry
+from threading import current_thread
+from pprint import pformat
 
 # WMCore modules
 from WMCore.MicroService.DataStructs.DefaultStructs import RULECLEANER_REPORT
+from WMCore.MicroService.DataStructs.DefaultStructs import MSRuleCleanerWflow
 from WMCore.MicroService.Unified.MSCore import MSCore
+from WMCore.MicroService.Unified.Common import ckey, cert
+from WMCore.Services.pycurl_manager import RequestHandler
+from WMCore.ReqMgr.DataStructs import RequestStatus
 from Utils.EmailAlert import EmailAlert
+from Utils.Pipeline import Pipeline, Functor
+from WMCore.WMException import WMException
+
+
+class MSRuleCleanerArchival(WMException):
+    """
+    Archival Exception Class for MSRuleCleaner Module in WMCore MicroServices
+    """
+    def __init__(self, message):
+        self.myMessage = "MSRuleCleanerArchival: %s" % message
```

Comment:
`myMessage` does not need to be an object attribute.

Reply:
Done